lib/ansible/modules/system/macos_pref.py

```diff
@@ -0,0 +1,1079 @@
+            .format(repr(value), type(value), given_type)
+        )
+
+    def run(self):
+
+        # Read the current value
+        self.current_value = self.return_value = self.prefs.read(self.key)
+
+        # Delete the key if state is "absent"
+        if self.state == "absent":
+            self.should_return_value = False
+            if self.current_value is None:
+                return
+
+            self.changed = True
+            self.return_value = None
+
+            if self.check_mode:
+                return
+
+            self.prefs.delete(self.key)
+
+        # Write if value is present
+        if self.value:
+            self.should_return_value = False
+            # Cast value type before comparing and writing.
+            if self.type:
+                self.value = self._cast_type(self.value, self.type)
+            else:
+                self.value = self._auto_cast_type(self.value)
+
+            # For state=merge, when base is array convert single item to a list.
+            if self.merge and isinstance(self.current_value, list) and not isinstance(self.value, list):
+                self.value = [self.value]
+
+            # Check if there's a type mismatch.
+            if (self.current_value is not None and
+                    not equivalent_types(self.current_value, self.value)):
+                raise MacOSPrefException(
+                    'New value type does not match current value type for key '
+                    '{0} ({1!r} {2} -> {3!r} {4}).'
+                    .format(
+                        self.key, self.value, type(self.value),
+                        self.current_value, type(self.current_value)
+                    )
+                )
+
+            # Check for changes and abort if ther are none
+            if self.merge:
+                if self.current_value is not None:
+                    # compare arrays
+                    if isinstance(self.current_value, list):
+                        new_items = [
+                            item for item in self.value
+                            if item not in self.current_value
+                        ]
+                        if not new_items:
+                            return
+                    # compare dicts
+                    else:
+                        new_items = [
+                            key for key, val in self.value.items()
+                            if key not in self.current_value
+                        ]
+                        if not new_items:
+                            return
+            elif self.current_value == self.value:
+                return
+
+            self.changed = True
+
+            if self.check_mode:
+                return
+
+            # Change/Create/Set given key/value for domain in defaults.
+            self.prefs.write(self.key, self.value, self.merge)
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            domain=dict(
```

Comment:
I prefer to see this in collapsed view, rather than expanded.
Also, if a parameter is not required, you don't need to add it explicitly.
(It's easier to see at a glance which parameters are required)